tests/regression/test_neg_rebase_sanity_checks.py

```diff
@@ -0,0 +1,113 @@
+
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, stored_exited_validators, _) = oracle_report_sanity_checker.reportData(count - 1)
+
+    assert stored_exited_validators == sum(reported_validators_values)
+
+def test_negative_rebase_correct_exited_validators_count_neg_rebase(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    reported_validators = exited_validators_count()
+
+    reported_validators_values = [value + 3 for value in reported_validators.values()]
+    oracle_report(cl_diff=-ETH(40000), stakingModuleIdsWithNewlyExitedValidators=list(reported_validators.keys()),
+                   numExitedValidatorsByStakingModule=reported_validators_values)
+
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, stored_exited_validators, _) = oracle_report_sanity_checker.reportData(count - 1)
+
+    assert stored_exited_validators == sum(reported_validators_values)
+
+def test_negative_rebase_correct_balance_neg_rebase(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    cl_decrease = ETH(40000)
+    oracle_report(cl_diff=-cl_decrease, exclude_vaults_balances=True)
+
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, _, cl_balance) = oracle_report_sanity_checker.reportData(count - 1)
+    assert cl_balance == cl_decrease
+
+    cl_decrease2 = ETH(30000)
+    oracle_report(cl_diff=-cl_decrease2, exclude_vaults_balances=True)
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, _, cl_balance) = oracle_report_sanity_checker.reportData(count - 1)
+
+    assert cl_balance == cl_decrease2
+
+
+def test_blocked_huge_negative_rebase(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    (_, cl_validators, cl_balance) = contracts.lido.getBeaconStat()
+
+    max_cl_balance = (INITIAL_SLASHING_AMOUNT_PWEI + INACTIVITY_PENALTIES_AMOUNT_PWEI) * ONE_PWEI * cl_validators
+    error_cl_decrease = cl_balance // 10    # 10% of current balance will lead to error
+    print("max_cl_balance", max_cl_balance)
+    with reverts(encode_error("IncorrectCLBalanceDecrease(uint256, uint256)", [error_cl_decrease, max_cl_balance])):
+        oracle_report(cl_diff=-error_cl_decrease, exclude_vaults_balances=True, silent=True)
+
+def test_negative_rebase_more_than_54_reports(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    reported_validators_values = exited_validators_count().values()
+    for _ in range(58):
```

Comment:
I am curious why 58? The test name suggests 54 reports.

Reply:
Well. It says more than 54, so just to be sure that reporting is not blocked on the bound of 54 rebases